### PR TITLE
doc/rbd: minor changes to the rbd man page

### PR DESCRIPTION
**Pull request by nbalacha**
Fixes typos and grammar for some commands. Adds
additional details for some commandds.

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [x] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

File: doc/man/8/rbd.rst [hunk 1]
```diff
@@ -564,12 +562,12 @@ Commands
...
   Promote all non-primary images within a pool to primary.
-  Every mirroring enabled image will promoted in the pool.
+  Every mirror enabled image in the pool will promoted.
 
 :command:`mirror pool status` [--verbose] [*pool-name*]
   Show status for all mirrored images in the pool.
-  With --verbose, also show additionally output status
-  details for every mirroring image in the pool.
+  With --verbose, also show additional output status
+  details for every mirror enabled image in the pool.
```
Comment (idryomov) [inline doc/man/8/rbd.rst:570]:
```suggestion
  details for every mirror-enabled image in the pool.
```

File: doc/man/8/rbd.rst [hunk 2]
```diff
@@ -523,7 +521,7 @@ Commands
 
 :command:`mirror pool demote` [*pool-name*]
   Demote all primary images within a pool to non-primary.
-  Every mirroring enabled image will demoted in the pool.
+  Every mirror enabled image in the pool will be demoted.
```
Comment (anthonyeleven) [inline doc/man/8/rbd.rst:524]: mirror-enabled?

File: doc/man/8/rbd.rst [hunk 3]
```diff
@@ -564,12 +562,12 @@ Commands
...
 :command:`mirror pool promote` [--force] [*pool-name*]
   Promote all non-primary images within a pool to primary.
-  Every mirroring enabled image will promoted in the pool.
+  Every mirror enabled image in the pool will promoted.
 
 :command:`mirror pool status` [--verbose] [*pool-name*]
   Show status for all mirrored images in the pool.
-  With --verbose, also show additionally output status
-  details for every mirroring image in the pool.
+  With --verbose, also show additional output status
```
Comment (anthonyeleven) [inline doc/man/8/rbd.rst:569]: s/also//
mirror-enabled

File: doc/man/8/rbd.rst [hunk 4]
```diff
@@ -625,7 +623,7 @@ Commands
   Set a limit for the number of snapshots allowed on an image.
 
 :command:`snap ls` *image-spec*
-  Dump the list of snapshots inside a specific image.
+  Dump the list of snapshots on a specific image.
```
Comment (anthonyeleven) [inline doc/man/8/rbd.rst:626]: I think "of" is better yet

File: doc/man/8/rbd.rst [hunk 5]
```diff
@@ -668,20 +666,22 @@ Commands
 :command:`trash ls` [*pool-name*]
   List all entries from trash.
 
-:command:`trash mv` *image-spec*
+:command:`trash mv` [--expires-at <expires-at>] *image-spec*
   Move an image to the trash. Images, even ones actively in-use by 
-  clones, can be moved to the trash and deleted at a later time.
+  clones, can be moved to the trash and deleted at a later time. Use
+  --expires-at to set the expiration time of an image after which
```
Comment (anthonyeleven) [inline doc/man/8/rbd.rst:672]: `--expires-at`

File: doc/man/8/rbd.rst [hunk 6]
```diff
@@ -668,20 +666,22 @@ Commands
...
 
 :command:`trash restore` *image-id*  
   Restore an image from trash.
 
-:command:`trash rm` *image-id* 
+:command:`trash rm` [--force] *image-id*
   Delete an image from trash. If image deferment time has not expired
-  you can not removed it unless use force. But an actively in-use by clones 
-  or has snapshots can not be removed.
+  it can be removed using --force. An image that is actively in-use by clones
```
Comment (anthonyeleven) [inline doc/man/8/rbd.rst:683]: backticks

File: doc/man/8/rbd.rst (hunk 2 above)
Comment (idryomov) [inline doc/man/8/rbd.rst:524]: If replacing "mirroring" with "mirror" here, I think a hyphen is in order:
```suggestion
  Every mirror-enabled image in the pool will be demoted.
```

File: doc/man/8/rbd.rst [hunk 7]
```diff
@@ -564,12 +562,12 @@ Commands
 
 :command:`mirror pool promote` [--force] [*pool-name*]
   Promote all non-primary images within a pool to primary.
-  Every mirroring enabled image will promoted in the pool.
+  Every mirror enabled image in the pool will promoted.
```
Comment (idryomov) [inline doc/man/8/rbd.rst:565]:
```suggestion
  Every mirror-enabled image in the pool will promoted.
```

File: doc/man/8/rbd.rst (hunk 4 above)
Comment (nbalacha) [reply doc/man/8/rbd.rst:626]: Agreed.

File: doc/man/8/rbd.rst (hunk 5 above)
Comment (nbalacha) [reply doc/man/8/rbd.rst:672]: Sorry, did not understand this comment.

Comment (idryomov) [reply doc/man/8/rbd.rst:672]: I think Anthony meant "add backticks around --expires-at" here.

File: doc/man/8/rbd.rst (hunk 2 above)
Comment (nbalacha) [reply doc/man/8/rbd.rst:524]: Done.

File: doc/man/8/rbd.rst (hunk 3 above)
Comment (nbalacha) [reply doc/man/8/rbd.rst]: Done.

File: doc/man/8/rbd.rst (hunk 5 above)
Comment (nbalacha) [reply doc/man/8/rbd.rst]: Done.

File: doc/man/8/rbd.rst (hunk 6 above)
Comment (nbalacha) [reply doc/man/8/rbd.rst]: Done.

File: doc/man/8/rbd.rst [hunk 8]
```diff
@@ -499,14 +499,12 @@ Commands
 :command:`mirror image disable` [--force] *image-spec*
   Disable RBD mirroring for an image. If the mirroring is
   configured in ``image`` mode for the image's pool, then it
-  can be explicitly disabled mirroring for each image within
-  the pool.
+  must be explicitly disabled for each mirrored image within the pool.
```
[the remark this answers is not in the document]
Comment (nbalacha) [reply doc/man/8/rbd.rst]: Done.

File: doc/man/8/rbd.rst [hunk 9]
```diff
@@ -499,14 +499,12 @@ Commands
...
 
 :command:`mirror image enable` *image-spec* *mode*
   Enable RBD mirroring for an image. If the mirroring is
   configured in ``image`` mode for the image's pool, then it
-  can be explicitly enabled mirroring for each image within
-  the pool.
+  must be explicitly enabled for each image within the pool.
```
[the remark this answers is not in the document]
Comment (nbalacha) [reply doc/man/8/rbd.rst]: Done.

File: doc/man/8/rbd.rst (hunk 1 above)
Comment (nbalacha) [reply doc/man/8/rbd.rst]: Done.

File: doc/man/8/rbd.rst [hunk 10]
```diff
@@ -668,20 +666,22 @@ Commands
 :command:`trash ls` [*pool-name*]
   List all entries from trash.
 
-:command:`trash mv` *image-spec*
+:command:`trash mv` [--expires-at <expires-at>] *image-spec*
   Move an image to the trash. Images, even ones actively in-use by 
-  clones, can be moved to the trash and deleted at a later time.
+  clones, can be moved to the trash and deleted at a later time. Use
+  --expires-at to set the expiration time of an image after which
+  it can be purged.
```
[the remark this answers is not in the document]
Comment (nbalacha) [reply doc/man/8/rbd.rst]: Done.

File: doc/man/8/rbd.rst (hunk 6 above)
[the remark this answers is not in the document]
Comment (nbalacha) [reply doc/man/8/rbd.rst]: Done.

File: doc/man/8/rbd.rst (hunk 2 above)
Comment (nbalacha) [reply doc/man/8/rbd.rst]: Done.

File: doc/man/8/rbd.rst (hunk 7 above)
Comment (nbalacha) [reply doc/man/8/rbd.rst]: Done.

File: doc/man/8/rbd.rst [hunk 11]
```diff
@@ -564,12 +562,12 @@ Commands
...
 :command:`mirror pool promote` [--force] [*pool-name*]
   Promote all non-primary images within a pool to primary.
-  Every mirroring enabled image will promoted in the pool.
+  Every mirror-enabled image in the pool will promoted.
 
 :command:`mirror pool status` [--verbose] [*pool-name*]
   Show status for all mirrored images in the pool.
-  With --verbose, also show additionally output status
-  details for every mirroring image in the pool.
+  With --verbose, show additional output status
```
Comment (idryomov) [inline doc/man/8/rbd.rst:569]:
```suggestion
  With ``--verbose``, show additional output status
```

**Review by idryomov — APPROVED** [review]
Otherwise LGTM!

**Comment by idryomov** [comment]
Please amend the commit title to match the title of the PR -- it needs to be start with "doc".

**Comment by zdover23** [comment]
> Please amend the commit title to match the title of the PR -- it needs to start with "doc".

I'll take care of it.

File: doc/man/8/rbd.rst (hunk 8 above)
[the remark this answers is not in the document]
Comment (zdover23) [reply doc/man/8/rbd.rst]: If the default is opt-in, it makes sense to me to remove it, but I'm not an expert in this matter. @idryomov?

[the remark this answers is not in the document]
Comment (idryomov) [reply doc/man/8/rbd.rst]: This sentence now reads
> If the mirroring is configured in ``image`` mode for the image's pool, then it must be disabled for each image individually.

and seems to provide an additional detail, so I'd keep it.

[the remark this answers is not in the document]
Comment (zdover23) [reply doc/man/8/rbd.rst]: > This sentence now reads
> 
> > If the mirroring is configured in `image` mode for the image's pool, then it must be disabled for each image individually.
> 
> and seems to provide an additional detail, so I'd keep it.

Roger that. It is a dynamic time for this PR and for this man page. I'll come back in a couple of hours and backport it.

File: doc/man/8/rbd.rst [hunk 12]
```diff
@@ -564,12 +562,12 @@ Commands
 
 :command:`mirror pool promote` [--force] [*pool-name*]
   Promote all non-primary images within a pool to primary.
-  Every mirroring enabled image will promoted in the pool.
+  Every mirror-enabled image in the pool will promoted.
```
[the remark this answers is not in the document]
Comment (zdover23) [reply doc/man/8/rbd.rst]: @nbalacha, I'll take care of it.

**Comment by zdover23** [comment]
https://github.com/ceph/ceph/pull/56255 - Squid backport
https://github.com/ceph/ceph/pull/56256 - Reef backport
https://github.com/ceph/ceph/pull/56257 - Quincy backport

**Pull request merged**